src/test/gen/bloom_gen.cpp

```diff
@@ -0,0 +1,59 @@
+#include <test/gen/bloom_gen.h>
+#include <test/gen/crypto_gen.h>
+
+#include <bloom.h>
+#include <math.h>
+
+#include <rapidcheck/gen/Arbitrary.h>
+#include <rapidcheck/Gen.h>
+#include <rapidcheck/gen/Tuple.h>
+#include <rapidcheck/gen/Predicate.h>
+#include <rapidcheck/gen/Numeric.h>
+#include <rapidcheck/gen/Container.h>
+
+/** Generates a double between [0,1) */
+rc::Gen<double> BetweenZeroAndOne()
+{
+    return rc::gen::map(rc::gen::arbitrary<double>(), [](double x) {
+        double result = abs(fmod(x, 1));
+        assert(result >= 0 && result < 1);
+        return result;
+    });
+}
+
+rc::Gen<unsigned int> Between1And100()
```

Comment:
nit: looks like this method is defined with the same name here and in `merkleblock_gen.h`, how about:
* applying `static` more liberally?
* renaming one or the other?

Reply:
So basically mark as `static` in merkleblock_gen.h and then include that header file and use it here?

Reply:
Well they have different implementations for different return results. Calling this `bloom_gen.cpp` implementation `static` will prevent it from being used outside this file, which would reduce the risk of conflicts and otherwise be appropriate as the method is only used here. I don't think you want to apply `static` in the header, as that will result in the function being compiled into every including translation unit. Naming this to not have a potential conflict would also be positive.